src/lib_json/json_reader.cpp

```diff
@@ -1878,15 +1873,29 @@ class OurCharReader : public CharReader {
 
 public:
   OurCharReader(bool collectComments, OurFeatures const& features)
-      : collectComments_(collectComments), reader_(features) {}
-  bool parse(char const* beginDoc, char const* endDoc, Value* root,
-             String* errs) override {
-    bool ok = reader_.parse(beginDoc, endDoc, *root, collectComments_);
-    if (errs) {
-      *errs = reader_.getFormattedErrorMessages();
+      : collectComments_(collectComments), reader_(features) {
+        this->_impl=std::unique_ptr<Impl>(new Impl(collectComments, features));
```

Comment:
Spaces around the `=` here.
But I think this assignment can be eliminated by calling a protected CharReader constructor.

Reply:
(line 1871) If the `impl` is recording the `collectComments`, then `OurCharReader` shouldn't have a redundant copy of it.  I think `collectComments_` and `reader_` can be removed from the `OurCharReader` class.